drivers/s3.go

```diff
@@ -112,10 +117,13 @@ func NewCustomS3Driver(host, bucket, region, accessKey, accessKeySecret string,
 	}
 	if os.awsAccessKeyID != "" {
 		creds := credentials.NewStaticCredentials(os.awsAccessKeyID, os.awsSecretAccessKey, "")
-		cfg := aws.NewConfig().WithRegion(os.region).WithCredentials(creds)
-		cfg = cfg.WithEndpoint(host)
-		cfg = cfg.WithS3ForcePathStyle(true)
-		os.s3svc = s3.New(session.New(), cfg)
+		cfg := aws.NewConfig().
+			WithRegion(os.region).
+			WithCredentials(creds).
+			WithEndpoint(host).
+			WithS3ForcePathStyle(true)
+		os.s3sess = session.New(cfg)
```

Comment:
nit: I know it's not strictly related to your PR, but `session.New(cfg)` is deprecated, so consider changing to `session.NewSession(cfg)`.